### ISSUE TITLE
Help needed to upgrade ogre package to v13.3.4

### ISSUE THREAD

**Issue by dotoscat** [msys2/MINGW-packages#11628]
I've already started to make [changes](https://github.com/dotoscat/MINGW-packages/commit/a872526902457d8f3eb99ee488ca4e7b02abcd20) to upgrade ogre package to v13.3.4 according to [ogre v13.3.4 notes](https://github.com/OGRECave/ogre/blob/master/Docs/13-Notes.md).

The problem when I do **makepkg-mingw** and appears this prompt:

```
  patching file RenderSystems/Direct3D9/src/OgreD3D9Plugin.cpp
  Hunk #1 FAILED at 33.
  1 out of 1 hunk FAILED -- saving rejects to file RenderSystems/Direct3D9/src/OgreD3D9Plugin.cpp.rej
  can't find file to patch at input line 17
  Perhaps you used the wrong -p or --strip option?
  The text leading up to this was:
  --------------------------
  |diff --git a/RenderSystems/Direct3D9/src/OgreMinGWSupport.cpp b/RenderSystems/Direct3D9/src/OgreMinGWSupport.cpp
  |index 4bc377ac09..9a6fef69d7 100644
  |--- a/RenderSystems/Direct3D9/src/OgreMinGWSupport.cpp
  |+++ b/RenderSystems/Direct3D9/src/OgreMinGWSupport.cpp
  --------------------------
  File to patch:
```

I don't know how to proceed. Please, help or tip of some kind.

**Comment by Biswa96** [comment]
> can't find file to patch at input line 17

There are many possibilities, like:

1. that file is moved to another location
2. the code is integrated into another file
3. that hunk is not required anymore
etc.

**Comment by Biswa96** [comment]
Wondering, do we really need that huge package?

**Comment by stahta01** [comment]
When trying to update a package to a new version, I comment out the patches that do not apply to see if it can build without the patch files that have issues. After it builds or fails to build I review each patch to see if the ones commented out need to fixed or removed.

**Comment by stahta01** [comment]
I tried to apply the patches and I am guessing that 020 patch is no longer needed.
And, I am thinking that patch 014 needs to be fixed to work with current code.
I have no idea if 011 needs fixed or can be skipped.
The other patch files applied to version "13.3.4" of the code.

Tim S.

**Comment by revelator** [comment]
yep no longer needed :) there is also this -> 012-add-link-library.patch which is also no longer required.
besides that the hack for imgui is also no longer needed as the package now downloads it to the right place correctly.

**Comment by revelator** [comment]
builds fine though i use ninja in my personal build as it seems to resolve paths a bit better.
setup looks about the same as my personal version as well, the security cookie stuff had me a bit baffled btw, as i could not find it 
anywhere in the newer sources. so maybe we need to readd this somewhere else as it seems there is some problems running the 32 bit sample programs. i cannot however say if this might be the cause but maybe something to look out for ?.

**Comment by revelator** [comment]
not the security cookie  stuff it seems, adding this elsewhere leads to a multiple definition error so it seems it was fixed in the mingw-w64 abi.

**Comment by revelator** [comment]
ok 64 bit samplebrowser works, the 32 bit version crashes so i tried debugging it and it throws an error about D3D9 overlay ?!?.

**Comment by revelator** [comment]
well seems the latest assimp no longer installs libIrrXml.a so if updating to that it will fail to build regardless.
in fact it seems IrrXml has been removed completely from assimp ?!?.

**Comment by revelator** [comment]
in case this was missed it builds and works just fine so can be committed :)

**Comment by dotoscat** [comment]
> in case this was missed it builds and works just fine so can be committed :)

oh nice!

**Comment by revelator** [comment]
anyone got stahta's patches for this somewhere ? my old workhorse for compiling bit the dust recently and waiting on a replacement mainboard for it. having a bit of a problem getting cmake to install the plugin import libraries in the correct place.

**Comment by MehdiChinoune** [comment]
ogre was updated to 13.4.4

**Comment by revelator** [comment]
cool thanks stahta :)
